tests.py

```diff
@@ -320,5 +323,25 @@ def test_equality(self):
         self.assertTrue(Symbol("db/id") == Symbol("db/id"))
 
 
+class ImmutableListTest(unittest.TestCase):
+    def test_list(self):
+        x = ImmutableList([1, 2, 3])
+        self.assertTrue(x == [1, 2, 3])
+
+        self.assertIsNone(x.append(4))
+        self.assertIsNone(x.extend(x))
+        self.assertIsNone(x.reverse())
+        self.assertIsNone(x.remove(1))
+
+        self.assertTrue(x.pop(0) == 1)
```

Comment:
You should use `assertEqual(1, x.pop(0))` here; it gives a better error message when it fails.